src/golibmc.go

```diff
@@ -73,23 +77,52 @@ var (
 	// Keys must be at maximum 250 bytes long, ASCII, and not
 	// contain whitespace or control characters.
 	ErrMalformedKey = errors.New("malformed: key is too long or contains invalid characters")
+
+	// ErrMemcachedClosed is returned when the memcached connection is
+	// already closing.
+	ErrMemcachedClosed = errors.New("memcached is closed")
+
+	// ErrBadConn is returned when the connection is out of its lifetime.
+	ErrBadConn = errors.New("bad conn")
 )
 
 func networkError(msg string) error {
 	return errors.New("libmc: network error(" + msg + ")")
 }
 
-// Default memcached port
+// DefaultPort memcached port
 const DefaultPort = 11211
 
 // Client struct
 type Client struct {
-	_imp        unsafe.Pointer
-	servers     []string
-	prefix      string
-	noreply     bool
-	disableLock bool
-	lk          sync.Mutex
+	servers        []string
+	prefix         string
+	noreply        bool
+	disableLock    bool
+	hashFunc       int
+	failover       bool
+	connectTimeout C.int
+	pollTimeout    C.int
+	retryTimeout   C.int
+	lk             sync.Mutex
+	freeConns      []*conn
+	numOpen        int
+	openerCh       chan struct{}
+	connRequest    map[uint64]chan *conn
+	nextRequest    uint64
+	maxLifetime    time.Duration // maximum amount of time a connection may be reused
+	maxOpen        int
```

Comment:
maybe should comment "// <= 0 means unlimited"

Reply:
Thanks, I added comment.
https://github.com/douban/libmc/pull/57/commits/f25d99cd8a6bca84367130d6e9e9212c3c042f4d